code/API_definitions/geofencing.yaml

```diff
@@ -0,0 +1,603 @@
+      description: The request for creating a Location Subscription
+      type: object
+      properties:
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+        subscriptionDetail:
+          $ref: '#/components/schemas/SubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The time when the location-tracking has to be terminated (e.g. in 14 days from now on). Provided in date-time format.
+      required:
+        - webhook
+
+    SubscriptionInfo:
+      description: Represents a area-location event subscription.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateSubscription'
+        - type: object
+          properties:
+            eventSubscriptionId:
+              $ref: '#/components/schemas/SubscriptionId'
+            startsAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking started (e.g. in 7 days from now on). Provided in date-time format.
+            expiresAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking expired (e.g. in 7 days from now on). Provided in date-time format.
+          required:
+            - eventSubscriptionId
+            - webhook
+
+    EventTime:
+      format: date-time
+      type: string
+      description: |
+        Timestamp of when the occurrence happened. 
+        If the time of the occurrence cannot be determined then this attribute MAY be set to some other time (such as the current time) by the CloudEvents producer, 
+        however all producers for the same source MUST be consistent in this respect. In other words, 
+        either they all use the actual time of the occurrence or they all use the same algorithm to determine the value used. 
+        (must adhere on CAMARA datetime recommendation based on RFC 3339)
+
+    EventNotifications:
+      description: The event notification callback.
+      type: object
+      required:
+        - id
+        - source
+        - type
+        - specversion
+        - time
+      properties:
+        id:
+          $ref: '#/components/schemas/EventId'
+        source:
+          description: |
+            identifies the context in which an event happened in the specific Provider Implementation. 
+            Often this will include information such as the type of the event source, the organization publishing the event or the process that produced the event. 
+            The exact syntax and semantics behind the data encoded in the URI is defined by the event producer.
+          type: string
+          example: "https://notificationSendServer12.supertelco.com"
+        type:
+          $ref: '#/components/schemas/EventType'
```

Comment:
Type (or eventType) should be tagged as discriminator and we should have something like this:
`      discriminator:
        propertyName: 'type'
        mapping:
            org.camaraproject.geofencing.v0....: '#/components/schemas/...'
            org.camaraproject.device-status.v0.subscription-ends: '#/components/schemas/EventSubscriptionEnds'`

As previously suggested I guess we could have separate event type for entered/exit.

Reply:
1.) How should EventSubscriptionEnds look like or could it be also part of the "Type"-schema:

```yaml
    Type:
      description: |
        org.camaraproject.geofencing.v0.area-entered - Location event triggered when the device enters the given area
        org.camaraproject.geofencing.v0.area-left - Location event triggered when the device leaves the given area
        org.camaraproject.device-status.v0.subscription-ends - Subscription ends for the given device
      type: string
      enum:
        - org.camaraproject.geofencing.v0.area-entered
        - org.camaraproject.geofencing.v0.area-left
        - org.camaraproject.device-status.v0.subscription-ends
```

2.) Shouldn't enums be in uppercase?
3.) How would the discriminator look like in this case?

```yaml
      discriminator:
        propertyName: 'type'
        mapping:
          org.camaraproject.geofencing.v0.area-left: '#/components/schemas/Type'
          org.camaraproject.geofencing.v0.area-entered: '#/components/schemas/Type'
          org.camaraproject.device-status.v0.subscription-ends: '#/components/schemas/Type'
```

I am not familiar with discriminators, but my example doesn't seem right

Reply:
Hi @maxl2287, in my review I point to a couple of examples from other APIs you can use